components/shortcut/actions/create-story/create-story.js

```diff
@@ -0,0 +1,357 @@
+          });
+        });
+        return options;
+      },
+      optional: true,
+    },
+    followerIds: {
+      type: "string[]",
+      label: "Follower Ids",
+      description: "A string array of UUIDs of the followers of this story.",
+      async options() {
+        return await this.shortcut.listMembersAsOptions();
+      },
+      optional: true,
+    },
+    iterationId: {
+      type: "integer",
+      label: "Iteration Id",
+      description: "The ID of the iteration the story belongs to.",
+      async options() {
+        const options = [];
+        const iterations = await this.shortcut.callWithRetry("listIterations");
+        iterations.forEach((iteration) => {
+          options.push({
+            label: iteration.name,
+            value: iteration.id,
+          });
+        });
+        return options;
+      },
+      optional: true,
+    },
+    label: {
+      type: "object",
+      label: "Label",
+      description:
+        "A label object attached to the story. Each label object must have the following structure: `color` which is an string with the hex color to be displayed with the Label i.e. \"#ff0000\", and a string `name` for the name of the Label. See [CreateLabelParams](https://shortcut.com/api/rest/v3#CreateLabelParams) for more info.",
+      optional: true,
+    },
+    linkedFileIds: {
+      type: "integer[]",
+      label: "Linked File Ids",
+      description:
+        "An integer array with the IDs of linked files attached to the story.",
+      async options() {
+        const options = [];
+        const linkedFiles = await this.shortcut.callWithRetry("listLinkedFiles");
+        linkedFiles.forEach((linkedFile) => {
+          options.push({
+            label: linkedFile.name,
+            value: linkedFile.id,
+          });
+        });
+        return options;
+      },
+      optional: true,
+    },
+    name: {
+      type: "string",
+      label: "Name",
+      description: "The name of the story.",
+    },
+    ownerIds: {
+      type: "string[]",
+      label: "Owner Ids",
+      description: "A string array of UUIDs of the owners of this story.",
+      async options() {
+        return await this.shortcut.listMembersAsOptions();
+      },
+      optional: true,
+    },
+    projectId: {
+      type: "integer",
+      label: "Project Id",
+      description: "The ID of the project the story belongs to.",
+      async options() {
+        const options = [];
+        const projects = await this.shortcut.callWithRetry("listProjects");
+        projects.forEach((project) => {
+          options.push({
+            label: project.name,
+            value: project.id,
+          });
+        });
+        return options;
+      },
+    },
+    requestedById: {
+      type: "string",
+      label: "Requested by ID",
+      description: "The ID of the member that requested the story.",
+      async options() {
+        return await this.shortcut.listMembersAsOptions();
```

Comment:
i'm sorry @jcortes , i am confused. shouldn't async functions must use await on their body?
I am 100% sure this is true for C#/.NET. Probably JS/Node is the same here. Please correct me if this isn't the case for javascript.

Reply:
Hi @sergioeliot2039 I've found [this article](https://jakearchibald.com/2017/await-vs-return-vs-return-await/) very interesting in order to totally understand `await` vs `return` vs `return await`. I hope it helps!

Reply:
oh, got it, after checking the article this falls in the case where`return await` is redundant, because it's outside a try/catch block. removing the await.